### PR TITLE
Fix deprecation warnings since FFmpeg 3.1

### PR DESCRIPTION
**Pull request by chiro**
Hi eagletmt,

I confirmed that this change doesn't affect the output ts with some videos. Any comments or suggestions are welcome!

### CODE REVIEW

File: clean-ts/clean-ts.c [hunk 1]
```diff
@@ -119,14 +119,18 @@ static int clean_ts(const char *infile, const char *outfile, int64_t npackets,
   output_streams = av_mallocz_array(input_stream_size, sizeof(*output_streams));
   size_t i;
   for (i = 0; i < input_stream_size; i++) {
-    output_streams[i] = avformat_new_stream(oc, input_streams[i]->codec->codec);
-    DPRINTF("%d: Copy from [0x%x]\n", output_streams[i]->index, input_streams[i]->id);
-    FAIL_IF_ERROR(avcodec_copy_context(output_streams[i]->codec, input_streams[i]->codec));
+    const AVCodec *codec = avcodec_find_encoder(input_streams[i]->codecpar->codec_id);
+    output_streams[i] = avformat_new_stream(oc, codec);
+    DPRINTF("%d: Copy from [0x%x]\n", output_streams[i]->index, input_streams[i]->index);
+    output_streams[i]->codecpar = input_streams[i]->codecpar;
```
Comment (eagletmt) [inline clean-ts/clean-ts.c]: Is copying pointer really safe? I've found `avcodec_parameters_copy` utility.

File: clean-ts/clean-ts.c [hunk 2]
```diff
@@ -119,14 +119,18 @@ static int clean_ts(const char *infile, const char *outfile, int64_t npackets,
...
     output_streams[i]->time_base = input_streams[i]->time_base;
   }
-  if (oc->oformat->flags & AVFMT_GLOBALHEADER) {
-    for (i = 0; i < input_stream_size; i++) {
-      output_streams[i]->codec->flags |= CODEC_FLAG_GLOBAL_HEADER;
+
+  for (i = 0; i < input_stream_size; i++) {
+    if (oc->oformat->flags & AVFMT_GLOBALHEADER) {
```
Comment (eagletmt) [inline clean-ts/clean-ts.c]: Why did you move this branching into the loop? The condition `oc->oformat->flags & AVFMT_GLOBALHEADER` doesn't change during the loop.

File: clean-ts/clean-ts.c [hunk 3]
```diff
@@ -119,14 +119,18 @@ static int clean_ts(const char *infile, const char *outfile, int64_t npackets,
...
+      output_streams[i]->codecpar->extradata_size = CODEC_FLAG_GLOBAL_HEADER;
```
Comment (eagletmt) [inline clean-ts/clean-ts.c]: `CODEC_FLAG_GLOBAL_HEADER` isn't size but bit flag.
I didn't find the correct field for setting this flag, but `extradata_size` looks wrong.

File: clean-ts/clean-ts.c [hunk 4]
```diff
@@ -172,7 +176,6 @@ static int clean_ts(const char *infile, const char *outfile, int64_t npackets,
   if (oc != NULL && oc->pb != NULL) {
     avio_close(oc->pb);
   }
-  avformat_free_context(oc);
```
Comment (eagletmt) [inline clean-ts/clean-ts.c:175]: Why did you delete this? `oc` is allocated by `avformat_alloc_output_context2` and it needs to be freed.

File: clean-ts/clean-ts.c (hunk 3 above)
Comment (eagletmt) [reply clean-ts/clean-ts.c]: Since `AVFMT_GLOBALHEADER` isn't set for MPEG2-TS, we could omit `CODEC_FLAG_GLOBAL_HEADER` .

File: clean-ts/clean-ts.c (hunk 1 above)
Comment (chiro) [reply clean-ts/clean-ts.c]: I wasn't aware of that... Changed to use it. Thanks.

File: clean-ts/clean-ts.c (hunk 2 above)
Comment (chiro) [reply clean-ts/clean-ts.c]: It's just for debugging. Sorry, I forgot to revert it.

File: clean-ts/clean-ts.c (hunk 3 above)
Comment (chiro) [reply clean-ts/clean-ts.c]: I deleted the loop. Thanks for investigating.

File: clean-ts/clean-ts.c (hunk 4 above)
Comment (chiro) [reply clean-ts/clean-ts.c:175]: oops, reverted.

**Comment by chiro** [comment]
Thanks for the quick review! I've addressed the your review comments. Please take a look.

**Comment by eagletmt** [comment]
Thanks a lot ❤️
